org/pr/ios-macos.ts

```diff
@@ -16,8 +16,9 @@ export default async () => {
 
     // Podfile should not reference commit hashes
     const podfileContents = await danger.github.utils.fileContents("Podfile");
-    const matches = podfileContents.match(/^[^#[]*:commit/gm);
-    if (matches !== null) {
+    const matchesOldRuby = podfileContents.match(/^[^#[]*:commit =>/gm);
+    const matchesNewRuby = podfileContents.match(/^[^#[]*commit:/gm);
```

Comment:
What is the point of excluding the `[` in `[^#[]`?

```suggestion
    const matchesOldRuby = podfileContents.match(/^[^#]*:commit =>/gm);
    const matchesNewRuby = podfileContents.match(/^[^#]*commit:/gm);
```

Reply:
Also I wonder if there's a risk of people using more spaces between `:commit` and `=>`, maybe a ` *` would be better?

Which makes me wonder… should we really lint the `Podfile` itself? Wouldn't it be more resilient to lint the `Podfile.lock` (which can be parsed as a `YAML` file) and detect pod referenced as commits from there, not depending on the syntax used by the user but instead relying on machine-generated counterpart?

Reply:
We can potentially use `be pod ipc podfile-json Podfile` for easier parsing?

Reply:
@AliSoftware:

> Which makes me wonder… should we really lint the `Podfile` itself? Wouldn't it be more resilient to lint the `Podfile.lock` (which can be parsed as a YAML file)

Ah, great idea. I went in and sort of patched the existing logic, but taking a step back reveals good improvements (good example of _local vs global maxima_).

I'll see what I can do on that front. The good thing is that the tests shouldn't need to change 😄

Reply:
@crazytonyli 

> We can potentially use `be pod ipc podfile-json Podfile` for easier parsing?

(I assume `be` is an alias for `bundle exec`–I use that, too)

I guess we could? However, Peril runs on Node and I'm tempted to try parsing the `Podfile.lock` YAML first, before trying to (I'm guessing) shell-out to run a Ruby command.

Reply:
> (I assume be is an alias for bundle exec–I use that, too)

Haha, yes indeed.

> Peril runs on Node

Oh, of course, sorry I thought this script runs on macOS.

Reply:
> I'm tempted to try parsing the `Podfile.lock` YAML first

See also https://github.com/wordpress-mobile/release-toolkit/issues/216 (where I left an example of how to do such parsing of pods list from the lockfile in Ruby)